service/src/main/java/uk/gov/hmcts/reform/fpl/handlers/FurtherEvidenceUploadedEventHandler.java

```diff
@@ -10,38 +10,45 @@
 import uk.gov.hmcts.reform.fpl.model.SupportingEvidenceBundle;
 import uk.gov.hmcts.reform.fpl.model.common.Element;
 import uk.gov.hmcts.reform.fpl.service.FurtherEvidenceNotificationService;
+import uk.gov.hmcts.reform.fpl.service.UserService;
 import uk.gov.hmcts.reform.idam.client.models.UserDetails;
 
 import java.util.HashSet;
 import java.util.List;
 import java.util.Objects;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 import static org.apache.commons.lang3.ObjectUtils.isNotEmpty;
+import static uk.gov.hmcts.reform.fpl.enums.CaseRole.representativeSolicitors;
 import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.unwrapElements;
 
 @Slf4j
 @Component
 @RequiredArgsConstructor(onConstructor = @__(@Autowired))
 public class FurtherEvidenceUploadedEventHandler {
     private final FurtherEvidenceNotificationService furtherEvidenceNotificationService;
+    private final UserService userService;
 
     @EventListener
     public void handleDocumentUploadedEvent(final FurtherEvidenceUploadedEvent event) {
         final CaseData caseData = event.getCaseData();
         final CaseData caseDataBefore = event.getCaseDataBefore();
         final UserDetails uploader = event.getInitiatedBy();
 
+        boolean uploadedByLA = event.isUploadedByLA();
+        List<Element<SupportingEvidenceBundle>> newBundle = getEvidenceBundle(caseData, uploadedByLA);
+        List<Element<SupportingEvidenceBundle>> oldBundle = getEvidenceBundle(caseDataBefore, uploadedByLA);
+        List<String> newNonConfidentialDocuments = getNewNonConfidentialDocuments(newBundle, oldBundle);
+
         final Set<String> recipients = new HashSet<>();
 
-        if (event.isUploadedByLA()) {
-            if (hasNewNonConfidentialDocuments(caseData.getFurtherEvidenceDocumentsLA(),
-                caseDataBefore.getFurtherEvidenceDocumentsLA())) {
+        if (uploadedByLA) {
```

Comment:
I would flip the ordering of these conditions, you are calling an if on `!newNonConfidentialDocuments.isEmpty()` twice so that could be the outer if and then have the `addAll` determined by `uploadedByLA`